### PR TITLE
(GH-476)(GH-477) Add Bolt Feature Support

### LINKED ISSUE #476: Bolt Command Palatte
- [ ] Open bolt user level config
- [ ] Open user level inventory config

### PR DESCRIPTION
**Pull request by jpogran**
- [x] Bolt open user config yml
- [x] Bolt open user inventory yml
- [x] Bolt yaml snippets for config and inventory
- [x] Readme examples

Fixes #476 and #477

### CODE REVIEW

File: src/feature/BoltFeature.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+              placeHolder: `Default bolt config yml not present. Do you want to create it?`,
+              canPickMany: false,
+              ignoreFocusOut: true
+            })
+            .then(answer => {
+              switch (answer) {
+                case 'no':
+                  break;
+                case 'yes':
+                  fs.writeFile(userInventoryFile, '', function(err) {});
```
Comment (glennsarti) [inline src/feature/BoltFeature.ts]: Do we need to specify UTF8 encoding here? or add boilerplate examples commented out?

File: package.json [hunk 2]
```diff
@@ -92,6 +93,20 @@
...
         ],
         "configuration": "./languages/puppetfile-language-configuration.json"
+      },
+      {
+        "id": "yaml",
+        "aliases": [
+          "YAML",
+          "yaml"
+        ],
+        "extensions": [
```
Comment (glennsarti) [inline package.json]: Given that these are only in `/tasks/*.yml` should this be a filename or path pattern?

File: languages/yaml-language-configuration.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
+{
```
Comment (glennsarti) [inline languages/yaml-language-configuration.json]: Not sure we need this given VSCode natively understands YAML

**Comment by glennsarti** [comment]
I don't think we need the Yaml language configuration or sytnax highlighting

File: src/extension.ts [hunk 4]
```diff
@@ -1,26 +1,26 @@
...
 import { IFeature } from './feature';
+import { BoltConfigFeature, BoltInventoryFeature } from './feature/BoltFeature';
 import { DebuggingFeature } from './feature/DebuggingFeature';
 import { FormatDocumentFeature } from './feature/FormatDocumentFeature';
 import { NodeGraphFeature } from './feature/NodeGraphFeature';
 import { PDKFeature } from './feature/PDKFeature';
 import { PuppetResourceFeature } from './feature/PuppetResourceFeature';
-import { ProtocolType, ConnectionType, IConnectionConfiguration } from './interfaces';
+import { ConnectionHandler } from './handler';
+import { DockerConnectionHandler } from './handlers/docker';
```
Comment (glennsarti) [inline src/extension.ts]: Do we need the individual handlers as imports? I can;t see where they get used

Comment (glennsarti) [reply src/extension.ts]: Doh...this is a move not add.  Ignore me.

File: src/feature/BoltFeature.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    );
+  }
+}
+
+export class BoltInventoryFeature implements IFeature {
+  dispose() {}
+  constructor(context: ExtensionContext) {
+    context.subscriptions.push(
+      commands.registerCommand('puppet-bolt.OpenUserInventoryFile', () => {
+        let userInventoryFile = path.join(process.env['USERPROFILE'] || '~', '.puppetlabs', 'bolt', 'inventory.yml');
```
Comment (glennsarti) [inline src/feature/BoltFeature.ts]: Docs and code say the file is called `inventory.yaml`

https://github.com/puppetlabs/bolt/blob/b2a3511bf04f7cf9b42eeddff215b6ddcf75765b/lib/bolt/boltdir.rb#L35

https://puppet.com/docs/bolt/1.x/inventory_file.html

File: src/feature/BoltFeature.ts [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+import { IFeature } from '../feature';
+import * as path from 'path';
+import * as fs from 'fs';
+
+export class BoltConfigFeature implements IFeature {
+  dispose() {}
+  constructor(context: ExtensionContext) {
+    context.subscriptions.push(
+      commands.registerCommand('puppet-bolt.OpenUserConfigFile', () => {
+        let userInventoryFile = path.join(process.env['USERPROFILE'] || '~', '.puppetlabs', 'bolt', 'bolt.yml');
```
Comment (glennsarti) [inline src/feature/BoltFeature.ts]: Docs and code say the file is called `bolt.yaml`

https://github.com/puppetlabs/bolt/blob/b2a3511bf04f7cf9b42eeddff215b6ddcf75765b/lib/bolt/boltdir.rb#L34

https://puppet.com/docs/bolt/1.x/configuring_bolt.html

File: src/feature/BoltFeature.ts (hunk 1 above)
Comment (jpogran) [reply src/feature/BoltFeature.ts]: UTF8 good idea.

Boilerplate good idea, but better in the snippets than here

File: src/feature/BoltFeature.ts (hunk 5 above)
Comment (jpogran) [reply src/feature/BoltFeature.ts]: 👍

File: src/feature/BoltFeature.ts (hunk 6 above)
Comment (jpogran) [reply src/feature/BoltFeature.ts]: 👍

File: languages/yaml-language-configuration.json (hunk 3 above)
Comment (jpogran) [reply languages/yaml-language-configuration.json]: It is gone

File: package.json (hunk 2 above)
Comment (jpogran) [reply package.json]: Removed it entirely, as there are other better yaml extensions

File: src/feature/BoltFeature.ts (hunk 1 above)
Comment (jpogran) [reply src/feature/BoltFeature.ts]: So I added utf8, then a short commented out line pointing people to use the snippets

**Pull request merged**